lib/iris/fileformats/_pyke_rules/fc_rules_cf.krb

```diff
@@ -1624,7 +1624,7 @@ fc_extras
         attr_units = get_attr_units(cf_coord_var, attributes)
 
         def cf_var_as_array(cf_var):
-            dtype = cf_var.dtype
+            dtype = iris.fileformats.netcdf.get_actual_dtype(cf_var)
```

Comment:
If you agree with [comment](https://github.com/SciTools/iris/pull/2975/files#r174818424), then make the call `_get_actual_dtype`